core/src/main/java/io/crate/types/SingleColumnTableType.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.types;
+
+import io.crate.Streamer;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * A type which represents a table retrieved by a subquery with one column.
+ *
+ * Currently only supports a single column but can be extended
+ * to hold an array of types of multiple columns.
+ */
+public class SingleColumnTableType extends DataType<Object[]> implements CollectionType, Streamer<Object[]> {
+
+    public static final int ID = 102;
+    private DataType<?> valueType;
+
+    public SingleColumnTableType(DataType<?> innerType) {
+        this.valueType = innerType;
+    }
+
+    SingleColumnTableType() {}
+
+    @Override
+    public int id() {
+        return ID;
+    }
+
+    @Override
+    public String getName() {
+        return valueType.getName() + "_table";
+    }
+
+    @Override
+    public Streamer<?> streamer() {
+        return this;
+    }
+
+    @Override
+    public Object[] value(Object value) {
+        if (value == null) {
+            return null;
+        }
+        return (Object[]) value;
+    }
+
+    @Override
+    public boolean isConvertableTo(DataType other) {
+        return this.valueType.isConvertableTo(other) ||
```

Comment:
this is wrong, it would mean you can convert this collection type to any of its inner types.
pls take a look at the implementation in ArrayType (the null check is also wrong)
It would probably make sense to extract a superclass and reuse methods